### DISCUSSION TITLE
PV/SG Ready

### DISCUSSION THREAD

**Discussion by cyberthom42** [cyberthom42/vaillant-arotherm-plus#11] [General]
Einige wollen sicherlich ihre PV-Anlage mit der WP koppeln. Wenn man keinen SMA-Wechselrichter hat, geht die nur über den SG/PV-Ready Kontakt der Arotherm. 
Ich habe dazu nun einiges experimentiert. Leider scheint es so, dass das Verhalten hier auch von dem in der Anleitung abweicht. Zudem aber das Thema PV-Überschuss sehr spärlich in den Dokumentationen behandelt wird. Ich werde hier noch einen Wiki-Eintrag erzeugen, der zum einen den Stand der Dokumentationen sammelt (ist an unterschiedlichen Stellen dokumentiert) und zum anderen die Erfahrungen widerspiegeln soll. 
Hier in der Diskussion würde ich gern die Erfahrungen austauschen. 
Was mir aufgefallen ist:

1. eigentlich sollte sobald der PV-Kontakt geschlossen ist der WW-Speicher aufgeladen werden. Laut Dokumentation sollte dies mit maximaler Leistung geschehen. Beides funktioniert nicht bzw. nur bedingt. Es wird die Aktion "Warmwasser schnell" getriggert, allerdings sieht man in den ebus-Statistiken, dass WW nicht geladen wird, so lange die Speichertemp höher als Wunschtemp - Hysterese ist. Hierzu würde ich noch eine separate Diskussion aufmachen, da dies nicht PV-spezifisch ist. 
2. Wenn dann der WW-Speicher geladen wird, dann nur mit der aktuell eingestellten Betriebsart, also bei Eco läuft der Kompressor nur mit 40 Umdrehungen/s .
3. In der Anleitung steht auch, dass die Speicherladung über den PV-Kontakt unabhängig vom Zeitprofil erfolgt, das zweifle ich aufgrund 1.), habe ich aber bisher nicht separat getestet
4. Zum Thema heizen steht in den Anleitungen nur, dass der Heizungspuffer geladen wird. Hier gibt es keinerlei Dokumentation zu Direktheizkreisen (also ohne separatem Heizungspuffer, sondern nur mit dem obligatorischen Reihenrücklauf Puffer). Ich beobachte, dass die Soll VL-Temp um 4-5 Grad angehoben wird, allerdings nur im `SetMode` Flag, `hc1ActualFlowTempDesired` wird nicht verändert. `z1ActualRoomTempDesired` wird auch nicht angehoben.

**Comment by chrizzzp** [comment]
So, habe jetzt endlich auch mal den PV-Kontakt meiner Arotherm Split ausprobiert (Anschluß wie in der Wiki beschrieben). Vielleicht noch als Ergänzungen für die Wiki meine Beobachtungen:

zu 1. Schließen des PV-Kontakts triggert noch nicht den Start der Wärmepumpe. Es braucht zusätzlich eine Wärmeanforderung, damit sie anspringt (entweder über Energieintegral oder Hysterese). Siehe folgende Info von Vaillant, die ich in einem PV-Forum gefunden habe:
https://www.photovoltaikforum.com/core/attachment/7892-sg-ready-w%C3%A4rmepumpen-pdf/
Wenn man mehr als einen Heizkreis konfigurieren kann (z.B. mit VR71), lässt sich zum Erzeugen der Wärmeanforderung z.B. auch einfach ein (virtueller) Heizkreis aktivieren, der geeigenete VL-Sollwerteinstellungen (z.B. Festwert-Regelung) hat. Das geht am Regler oder auch per ebusd  (Hc2, Hc3...).

zu 4. Die Soll-VL-Temperatur (`hmu` `setmode`) der WP wird bei Schließen des Kontaktes genau um den Wert 'Offset Pufferspeicher' erhöht (im Regler VRC720/2 unter  Einstellungen → Fachhandwerkerebene → Anlagenkonfiguration). Default sind 10 K.
Per ebus kann man den Offset auch setzen, z.B. über `15.ctlv2.csv`:

```
*r,,,,,,B524,02000000,,,IGN:4,,,
*w,,,,,,B524,02010000,,,,,,
r;w,,CylinderOffset,Offset Pufferspeicher,,,,A200,,,calibrationv,,,Offset Pufferspeicher PV Ladung
```
Fällt der PV-Kontakt wieder ab (=offen), dann wird auch der Offset Pufferspeicher sofort wieder zurückgenommen. Gibt es ggf. dadurch keine Wärmeanforderung mehr, geht die WP aus.

Wie sinnvoll das ganze für die effiziente Nutzung von PV-Überschuß ist, darüber lässt sich streiten. Wenn man eine Steuerung über ebusd hat, braucht man den Kontakt eigentlich nicht. Die gleiche VL-Erhöhung kann man z.B. auch über das Erhöhen von `Hc1ExcessTemp` erreichen.

**Reply by xerion3800** [reply]
> Wenn man mehr als einen Heizkreis konfigurieren kann (z.B. mit VR71), lässt sich zum Erzeugen der Wärmeanforderung z.B. auch einfach ein (virtueller) Heizkreis aktivieren, der geeigenete VL-Sollwerteinstellungen (z.B. Festwert-Regelung) hat. Das geht am Regler oder auch per ebusd (Hc2, Hc3...).

Hast du das bei dir am Laufen bekommen?
Ich habe das auch vor längeren mal versucht das klappt leider bei einer aroTHERM Plus mit VR71 und VR720  nicht so ohne weiteres. Die Anleitung ist auch noch sehr alt also von einer aroTHERM (ohne Plus) und dem alten Regler VRC700. Ich warte auch schon seit ein paart Wochen auf eine qualifizierte Antwort von Vaillant aber die erste Antwort die kam war einfach Standard Bla Bla. Ich will das Nutzen wenn der dynamische Strompreis im Keller ist um die WP zu triggern falls mal kein Heizkreis läuft. Denn wie die auch schon geschrieben hast, funktioniert PV-Ready beim Pufferspeicher nur bei einer Heizungsanforderung. Falls man durch die Nachtabsenkung keine Heizungsanforderung hat dann wir der Puffer auch nicht "überladen". Dafür sollte dann der Modus 4 vom SG-Ready gut sein aber das klappt bei mir so nicht. Ich habe das aktuell über ebus selber gelöst und schalte dann den virtuellen HK an wenn HK1 z.B. aus sein sollte.

**Reply by chrizzzp** [reply]
Nein, ich habe die Heizanforderungen bisher immer manuell am Regler oder per ebus ausgelöst.

Ich kann mir nicht vorstellen, daß es bei der Arotherm Plus nicht geht. Gehört ja zum SG/PV-Ready-Standard. Welche Konfiguration des VR71 benutzt Du?

Siehe hier (VR71):  https://www.heizungsforum.de/attachments/1670440394455-png.32597

Für Zustand 4 werden in o.g. Fall die Öffner von K1 und K2 mit dem Anschluß S8 des VR71 verbunden.  In diesem Fall wäre also Öffnung von S8 eine Heizanforderung an HK3 (gilt aber m.E. nur für VR71-Konfiguration 3!). Im Regler (VRC720/2) kann man einstellen, wie die Heizanforderung interpretiert werden soll (Öffnen oder Schließen des Kontaktes, Alagenkonfiguration -> Konf. ext. Eingang).

In Konfiguration 1 (VR71) müssten die Kontakte mit S9 verbunden werden (HK2) und in Konfiguration 6 an S10 oder S11 (siehe Anleitung VRC720/2).

**Reply by xerion3800** [reply]
> Ich kann mir nicht vorstellen, daß es bei der Arotherm Plus nicht geht. Gehört ja zum SG/PV-Ready-Standard. Welche Konfiguration des VR71 benutzt Du?

Ich nutze folgende Konfig:
![image](https://github.com/cyberthom42/vaillant-arotherm-plus/assets/46169403/e3d0ef1e-9b6b-46f0-b8f8-f695764a7fd7)

Aktuell ist das so angeklemmt bei mir:
S21 für den SG-Ready Kontakt in der Hydraulikstation VWZ MEH 97/9
X41 für den PV-Ready Kontakt in der Hydraulikstation VWZ MEH 97/9

**Reply by chrizzzp** [reply]
Bei VR71 Konfiguration 3 müsste die Heizanforderung für Hk3 an S8 des VR71 anliegen (default ist Kontakt offen = Heizanforderung). Sprich, wenn der Kontakt bereits offen ist und Du im Regler mal Heizkreis 3 aktivierst (z.B. auf Festwert-Regelung, und die Soll-Temperatur-Einstellungen passen), sollte er direkt mit Heizen beginnen (es sei denn der externe Anschluss ist im Regler genau andersrum konfiguriert, also Kontakt gebrückt = Heizanforderung, in diesem Fall mal S8 brücken).

**Reply by xerion3800** [reply]
Ja stimmt habe das gerade mal getestet. Was nur blöd ist, das die "Brücken Einstellung" dann für alle HK gilt. Wenn ich also auf "Offen" stelle dann geht der HK1 auch nicht mehr. Habe da jetzt unter S6 eine Kabelbrücke reingemacht damit der HK1 immer funktioniert und wenn ich S8 dann per Relais brücke, startet die Festwertregelung vom HK3. Soweit erstmal gut, ich beobachte das mal. Danke erstmal für deine Unterstützung.

**Comment by cyberthom42** [comment]
Meine Versuche sind ja nun schon eine Weile her. Mein Fazit war, dass es absolut unbrauchbar ist. Es gibt ja nur einen relevanten Kontakt. Der zweite ist ja für die Steuerung des Netzbetreibers. Allerdings habe ich nur mit Zustand 3 probiert. Den Zustand 4 habe ich nicht probiert. Ich habe einen Pufferspeicher für WW (Trinkwasserstation) und einen Direktheizkreis mit Reihenrücklaufspeicher. Ich bin mir nicht sicher, ob hier Zustand 4 gehen würde. 
Letztendlich ist die Steuerung über den Kontakt eher schwierig. Wir betrachten ja hier die Monate im Winter oder im Übergang. Da schwankt die Sonneneinstrahlung immer sehr stark. Das heißt, es kann zu häufigen ein- und ausschalten führen. Gerade beim Ausschalten wird das Energieintegral dann sehr schnell steigen und die WP abschalten. 
Bei WW ist es so, dass die Speicehrladung dann nicht mehr abgebrochen wird. Also wenn für ein paar Minuten PV-Überschuss da ist und danach flaute, lädt er den WW-Speicher trotzdem fleißig weiter. 
Prinzipiell wird ja immer zuerst WW zubereitet (Zeitpunkt muss aber außerhalb der WW-Zeit liegen). Die kommt dann der Funktion "Warmwasser schnell" gleich. 
Aktuell fahre ich mit Zeitprogramm besser (WW z.b. immer 12:30). Die VL fahre ich morgens in 0,5 oder 1K schritten langsam hoch. Nachts auf ECO mit AT-Durchheizen auf 4 Grad eingestellt (Haus ist nicht so super gut gedämmt).

**Comment by cyberthom42** [comment]
Das einzige was Sinn macht ist eine Steuerung nach Solar-Prognose. Ich habe bei mir im Grafana schon seit Monaten zwei Prognose-Quellen drin. Zumindest forecast.solar ist ziemlich genau bezogen auf die nächsten Stunden. Hier reicht ja eine Vorhersage für die nächsten Stunden.

**Comment by chrizzzp** [comment]
Vaillant sieht die PV-Unterstützung bei der Heizung offensichtlich nur darin, den Vorlauf anzuheben um den Pufferspeicher etwas mehr aufzuheizen. Wie sinnvoll das ist, ist eine andere Frage (sofern man überhaupt einen Pufferspeicher hat). Da ich an meiner Heizung einen recht großen Pufferspeicher verbaut habe (da Kombination von FBH mit WP und wasserführender Kamin) und die WP (leider) überdimensioniert ist, kann man das tatsächlich auch einigermaßen nutzen und den PV-Überschuss in vergleichsweise kurzer Zeit in Wärme speichern. Kein Nachteil ohne Vorteile... ;-)

Ich würde die Ansteuerung der Kontakte trotzdem auch nicht "ungeglättet" aus dem PV-Überschuss ableiten (d.h. direkt aus dem Energiemanager o.ä.), sondern m.E. muß man sich eine (einigermaßen) "intelligente" Regelung überlegen und mit seiner Haus-Automatisierung umsetzen, um das Takten zu vermeiden. Also z.B. zumindest eine Art gleitenden Durchschnitt des PV-Überschusses über einen bestimmten Zeitraum nutzen, und damit Ein- und Ausschaltregeln definieren, die Mindestlaufzeiten oder ggf. auch Sperrzeiten abbilden. Ich bin da noch am Probieren... 
Das ist natürlich Aufwand. Man hört auch häufig, daß eine einfache Zeitsteuerung auch schon ganz gut funktioniert. Die PV-Vorhersage einzubinden, ist auch eine gute Ergänzung (habe ich auch), es kommt sicher auch etwas darauf an wie groß die PV-Anlage ist, d.h. der Ertrag relativ zum "Grundverbrauch" des Hauses.

**Reply by xerion3800** [reply]
PV Anlage habe ich auch aber aktuell geht es mir eher bei günstigen Preisen an der Börse. Ich nutze Tibber und bei extrem teuren Preise sperre ich die Pumpe für maximal zwei Stunden. Warmwasser mache ich immer Nachts ab 03 Uhr da dort der Preis am günstigsten ist. Wenn es mal wieder richtig günstig ist  (so wie seit drei Tagen so um die 13 Cent) würde ich dann auch noch den Pufferspeicher (100L) anheben und das geht ja aktuell nur bei aktiver Heizanforderung. Ich kann das auch per ebusd nachbilden wollte nur vermeiden mir alles selber nachzubauen wenn das mit einem einfachen Potentialfreien Kontakt möglich ist.

**Reply by cyberthom42** [reply]
Habt ihr das nun mal getestet, ob man damit auch eine Heizanforderung erzwingen kann?

**Reply by xerion3800** [reply]
Ja das funktioniert bei mir. 
Ich habe also physikalisch einen Heizkreis der läuft ganz normal über das Zeitfenster. Dann habe einen zweiten virtuellen Heizkreis (HK3) erstellt der mit Festwert auf 55°C eingestellt ist. 
Dann habe ich im VR71 für den HK1 eine Brücke reingemacht und für den HK3 mein Relais angeklemmt was dann bei Anforderung geschlossen wird.
Da ich einen Pufferspeicher für die Heizung habe, kann ich diesen dann "überfahren" Der Mischer für den HK1 regelt dann passend auf die VL-Temp die laut Heizkurve benötigt wird obwohl das Wasser im Puffer deutlich höher ist. Da für den HK3 kein Sensor verbaut ist der die Temperatur misst würde der HK3 erst dann wieder abschalten wenn das Relais ausgeschaltet wird oder das EI abgebaut wurde was dann sehr lange dauern kann. Ich vergleiche dann die Temperatur im Puffer mit einer Hysterese und schalte das Relais bei überschreiten der Hysterese wieder ab.  Wie schön erwähnt nutze ich das hauptsächlich um bei sehr günstigen Strompreisen diese in Form von Wärme zu speichern.
Im Estrich kann ich leider nicht so viel speichern da wir hauptsächlich Heizkörper haben mit teilweise Fussbodenerwärmung.

**Reply by chrizzzp** [reply]
100 L Pufferspeicher auf 55°C bringen reicht sicherlich nicht sonderlich lange in diesen Tagen, oder? ;-) 

Off-topic: braucht man für Tibber eine bestimmte Technik im Verteilerschrank (SmartMeter, von außen auslesbar) ?

**Reply by xerion3800** [reply]
Nein in diesen Tage nicht wirklich aber dann ist der Strom Erfahrungsgemäß auch nicht besonders günstig das ich den Puffer überlade. Wenn es milder ist sind die Preise nachts auch deutlich günstiger und dann komme ich mit der Erhöhung zumindest mit der Absenkung durch die Nacht und reduziere das Takten.

Bei Tibber gib es zwei Möglichkeiten, einmal die monatliche Abrechnung und die stündliche. Ich nutze die stündliche dafür braucht man ein SmartMeter oder wenn man einen Zweirichtungszähler durch  PV schon hat, dann reicht der Tibber-Pulse (den nutze ich auch).
Der kostet normalerweise um die 100 Euro aber wenn ich dich werbe bekommst du 50 Euro für den Shop und ich 50 Euro für den Shop. So bekommst du den Pulse dann für den halben Preis.
https://invite.tibber.com/t9yskmvd

**Reply by chrizzzp** [reply]
Noch eine Ergänzung zu meinen Beobachtungen den PV-/SG-ready-Kontakt betreffend:

Der 'Offset Pufferspeicher', um den die VL-Solltemperatur der WP bei Schließen des Kontaktes erhöht wird, ist auf maximal 15 K begrenzt.

Wenn man die Möglichkeit hat (mehrere Heizkreise mit VR71/FM5), dann ist es flexibler einen weiteren (virtuellen) Heizkreis zu nutzen, um die VL-Erhöhung zu erreichen (wie oben schon beschrieben). Hier hat man nicht die Begrenzung des 15K Offset, sondern kann einen Festwert für den Vorlauf frei definieren, der erreicht werden soll.

Per Hardware (Relais und Einstellungen am Regler) geht es wie oben beschrieben über die Heizanforderung (PV/SG Zustand 4) und per Software (ebusd) über die Heizkreiseinstellungen laut `ctlv2.csv` indem man den `CircuitType` auf '2' (Festwert) setzt (Ausschalten wieder auf '0' setzen). Komischerweise muß man auch die Solltemperatur für den virtuellen Heizkreis bei jeder Aktivierung neu setzen. Die wird irgendwie nach Deaktivieren des Heizkreises immer auf 65°C zurückgesetzt. Weiß noch nicht genau, wo der Default definiert wird.

Für Heizkreis 2 laut `ctlv2.csv` wären das folgende Definitionen:

```
*r,,,,,,B524,02000201,,,IGN:4,,,
*w,,,,,,B524,02010201,,,,,,
r;w,,Hc2CircuitType,CircuitType Heizkreis 2,,,,0200,,,mctype;IGN:1,,,0=off 2=fixed
r;w,,Hc2DesiredTemp,Solltemperatur Heizkreis 2,,,,0C00,,,tempv,,,

```
Die Leistungsregelung der WP bleibt davon m.E. unbeeinträchtigt, d.h. Energieintegral und Hysterese-Einstellung der WP werden beachtet. Der virtuelle Heizkreis führt also nicht zu einer unlimitierten Heizanforderung. Sinnvoll nutzbar ist das daher auch nur, wenn man die Wärme schnell genug weg bekommt, also z.B. in einen Heizungspufferspeicher oder WW aufheizt.
